Source/WebKit/NetworkProcess/soup/NetworkDataTaskSoup.cpp

```diff
@@ -55,6 +56,13 @@ using namespace WebCore;
 
 static const size_t gDefaultReadBufferSize = 8192;
 
+Ref<NetworkDataTask> NetworkDataTaskSoup::create(NetworkSession& session, NetworkDataTaskClient& client, const NetworkLoadParameters& parameters)
+{
+    if (parameters.request.url().protocolIsData())
+        return NetworkDataTaskDataURL::create(session, client, parameters);
```

Comment:
I think NetworkDataTask::create is better place to add this code. WDYT?
~~~
#if PLATFORM(COCOA)
    return NetworkDataTaskCocoa::create(session, client, parameters);
#else
    if (parameters.request.url().protocolIsData())
        return NetworkDataTaskDataURL::create(session, client, parameters);
~~~

https://github.com/WebKit/WebKit/blob/a0dcbd369295e3a93b80cbc22b0988678317c18f/Source/WebKit/NetworkProcess/NetworkDataTask.cpp#L58

Reply:
Done.